Libraries/LibJS/Runtime/GetterSetterPair.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2020, Matthew Olsson <matthewcolsson@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <LibJS/Runtime/Cell.h>
+#include <LibJS/Runtime/Value.h>
+
+namespace JS {
+
+class GetterSetterPair : public Cell {
+public:
+    static GetterSetterPair* create(Interpreter& interpreter, Value getter, Value setter)
+    {
+        return interpreter.heap().allocate<GetterSetterPair>(getter, setter);
+    }
+
+    GetterSetterPair(Value getter, Value setter)
+        : m_getter(getter)
+        , m_setter(setter)
+    {
+    }
+
+    Value& getter() { return m_getter; }
+    Value& setter() { return m_setter; }
+
+    Value call_getter(Value this_object)
+    {
+        if (getter().is_undefined())
```

Comment:
Instead of storing `undefined` in `m_getter` and `m_setter`, let's just store empty values.
That allows us to distinguish between empty (engine-only) and `undefined` (producible by JS.)

Reply:
My thought process here was that, since the only types of values that are valid for getters and setters are `undefined` and functions, that I should store them only as one of those two types. Otherwise, in the places that I currently check for them not being undefined, I'd have to check for both empty and undefined. 
Which of course isn't really a problem, it only happens in five or so places, but I'm not sure that we would really ever need to distinguish between empty and undefined values for Accessors.

Reply:
Seems like the thing we should really be checking is `is_function()`, no? :)

Reply:
Ehh, yup... that makes a lot more sense :)